gnucash/report/report-system/html-document.scm

```diff
@@ -391,7 +392,8 @@
                (lambda (obj doc)
                  (gnc:html-document-render-data doc obj))
                ;; if the object is #f, make it a placeholder
-               (if obj obj " ")))
+               ;;(if obj obj " ")))
+               obj))
```

Comment:
This change causes:
The following tests FAILED:
	 93 - test-invoice (Failed)
	107 - test-transaction (Failed)
	108 - test-balsheet-pnl (Failed)
	109 - test-income-gst (Failed)

With an error 
```
Throw to key `parser-error' with args `((#<input: string 10b506f50> "[GIMatch] b
roken for " (END . td) " while expecting " END boolean))'.
```

Reply:
yepp, I can also reproduce it on my end.
After looking into this a bit, it seems that #f was used as value for "empty" cell. If this is changed to "" then the test is passed. I have not found all places yet that use #f for empty cells to pass all tests. I keep looking into this.

Apart from this:
The xml->sxml converter doesn't like the #f value as only value in cells if they are used with the <boolean> tag, because it expects a closing tag that is not implemented.
Actually, that also should go for the <string> tag.

Enabling the <boolean> tag for #f and correcting the append-colomn! to produce more empty cells intensifies the effect on the failing reports.

I try to come up with a proposal, if it is small enough. Otherwise it might be more clever to simply not apply the changes for 796831 and 796821.

Reply:
As said before, I do not see a good way to go around this.
According to my understanding we need an additional condition that makes it possible to decide between
#f -> <boolean>#f
and
#f -> " "

The change above is in
html-document.scm: (gnc:make-html-object obj)

which is used by
html-document.scm: (gnc:html-object-render obj doc)
html-document.scm: (gnc:html-document-add-object! doc obj)

both of them would require a change of interface (some additional arg, I guess).
but (gnc:html-document-add-object! doc obj) is spread, so here we would need to touch a couple of report implementations, too.

Only way I see:  simply revert back and stick to how it has been up to now.
As you might guess, that is not my favourite.
BTW: I also had to rework my custom reports for new time handling. Writing own reports bears a risk that  it might require maintenance ...

Suggestions/Advice welcome.

Reply:
Maybe we're thinking of this incorrectly: What is the actual use-case for putting a "#t" or "#f" in a report? Reports are for humans, so rather than outputting a raw boolean object one should output the appropriate (localized) word for the situation, e.g. "True"/"False" or "Yes"/"No".
In most other cases we're using ```#f``` to mean "nothing to display" and the conversion to the empty string is the right thing to do.

Reply:
> Maybe we're thinking of this incorrectly: What is the actual use-case for putting a "#t" or "#f" in a report? Reports are for humans, so rather than outputting a raw boolean object one should output the appropriate (localized) word for the situation, e.g. "True"/"False" or "Yes"/"No".
> In most other cases we're using `#f` to mean "nothing to display" and the conversion to the empty string is the right thing to do.

I think this is the right answer. It is a very lispy convention that `#f` means default, or absence of an object. I do not think there's a need for appending boolean objects into a report at all.

Report objects are gnc:html-table, html-text, html-barchart, etc. From my understanding of old code, raw string or numbers are handled separately, stemming from legacy reports that have never been upgraded. Raw numbers are rendered with 2 decimal points, and strings are rendered raw. Unknown objects added to the document are rendered with invalid xml.

IMHO invalid objects should produce an error, which will signal the report writer to send proper document objects. We can do any report without resorting to coding boolean rendering separately, and without needing adjustments to current reports. I look forward to an amended PR incorporating tests... I will be very happy to refactor html-document which will explicitly clarify and document all acceptable document objects.